CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -34,6 +34,7 @@ PuppiPhoton::PuppiPhoton(const edm::ParameterSet& iConfig) {
   usePhotonId_           = (iConfig.getParameter<edm::InputTag>("photonId")).label().size() != 0;
```

Comment:
consumes for "photonName" ( tokenPhotonCandidates_ ) is needed only if usePFphotons_ is false.
Please do not declare consumes for the product which is not needed.